From 4571e0957117754566bbac007a31063709803607 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 05 Sep 2021 12:49:17 +0300
Subject: [PATCH] [SW-0] gaudi2: arc: fix the config base

Gaudi2 ARC maps 16 regions to access SoC.
The region 15 (16th) is the LBW (CFG) access, but is mapped to the
lower 128MB of the section, thus bit 27 should be clear.
The reason it works with bit 27 set is that h/w discards bit 27 when
ARC accesses CFG space, so effectively this bit is "don't care".
Yet, clear bit 27 so we have a clear and correct picture of the
address accessed.

Change-Id: I289d1d27757082c12c059e045c5b619161bf8526
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 2a55053..d616396 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -36,7 +36,7 @@
 
 #define SDRAM_BASE		0x11000000
 #define HL_SPI_BASE		0x26000000
-#define HL_CFG_BASE		0xF8000000
+#define HL_CFG_BASE		0xF0000000
 #define I2C_SLAVE_DEFAULT_INTR_MASK	0x48FF
 
 #define FUSE_WORDS_PER_BANK	0x40
